### PR TITLE
Configure CI

### PR DESCRIPTION
**Pull request by plafer**
Note: builds on #77 to reduce merge conflicts

Adds to the CI:
- nightly fmt
- clippy
- build docs
- run tests

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+          profile: minimal
+          toolchain: nightly
+          components: clippy
+          override: true
+
+      - name: Clippy
+        uses: actions-rs/cargo@v1
+        with:
+          command: clippy
+          args: --workspace --all-targets --all-features -- -D clippy::all -D warnings
```
Comment (hackaugusto) [inline .github/workflows/ci.yml:46]: we should also check using:

- `--all-targets --no-default-features`
- `--all-targets`

e.g. https://github.com/0xPolygonMiden/miden-vm/blob/main/.github/workflows/ci.yml#L18

**Review by hackaugusto — COMMENTED** [review]
note:

I'm 100% in favor of using rust-fmt, IMO we should enable that over board. However @bobbinth may have a different opinion.

We should also run clippy and cargo check using different feature flags, I left an inline comment a link to the miden-vm ci configuration to illustrated it.

**Comment by plafer** [comment]
@hackaugusto see https://github.com/0xPolygonMiden/miden-vm/pull/1158

File: block-producer/src/block_builder/errors.rs [hunk 2]
```diff
@@ -2,9 +2,8 @@ use miden_objects::accounts::AccountId;
 use miden_vm::{crypto::MerkleError, ExecutionError};
 use thiserror::Error;
 
-use crate::store::{ApplyBlockError, BlockInputsError};
-
 use super::prover::CREATED_NOTES_TREE_INSERTION_DEPTH;
+use crate::store::{ApplyBlockError, BlockInputsError};
 #[derive(Debug, Error, PartialEq, Eq)]
```
Comment (bobbinth) [inline block-producer/src/block_builder/errors.rs:7]: I think there should be a blank line between these two lines.

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you! I left one small comment inline.

**Comment by plafer** [comment]
@hackaugusto can I merge this into your branch?

File: block-producer/src/block_builder/errors.rs (hunk 2 above)
Comment (plafer) [reply block-producer/src/block_builder/errors.rs]: done

**Comment by hackaugusto** [comment]
> @hackaugusto can I merge this into your branch?

sure :)

**Pull request merged**